### PR TITLE
[BUILD/TEST] Formalize tests, project structure

### PR DESCRIPTION
**Pull request by tqchen**
- Add travis tests to make sure the style and compilation pass
- Formalize project structure to include stable components in include and others in contrib

### CODE REVIEW

File: include/dlpack/dlpack.h [hunk 1]
```diff
@@ -3,15 +3,18 @@
...
+#define DLPACK_DLPACK_H_
 
 #ifdef __cplusplus
 #define DLPACK_EXTERN_C extern "C"
 #else
 #define DLPACK_EXTERN_C
 #endif
 
+/*! \brief The current version of dlpack */
+#define DLPACK_VERSION 010
```
Comment (edgarriba) [inline include/dlpack/dlpack.h]: version number should go accord to
https://github.com/tqchen/dlpack/blob/master/CMakeLists.txt#L11

Comment (tqchen) [inline include/dlpack/dlpack.h]: Updated in the cmake list. is that good?

Comment (edgarriba) [inline include/dlpack/dlpack.h]: yeah, but not sure what's the proper way to have a centralized version registry

Comment (tqchen) [inline include/dlpack/dlpack.h]: AFIK we have to do it in both side

Comment (tqchen) [inline include/dlpack/dlpack.h]: @edgarriba Now that this is resolved, can we go ahead and merge this in?

**Pull request merged**